src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs

```diff
@@ -1,82 +1,263 @@
-﻿using Newtonsoft.Json;
+﻿using Microsoft.CodeAnalysis;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 using Stryker.Core.Initialisation.ProjectComponent;
 using Stryker.Core.Mutants;
 using Stryker.Core.Options;
 using System.Collections.Generic;
 using System.IO;
+using System.IO.Abstractions;
 using System.Linq;
 
 namespace Stryker.Core.Reporters
 {
     public class JsonReporter : IReporter
     {
         private readonly StrykerOptions _options;
-        public JsonReporter(StrykerOptions options)
+        private readonly IFileSystem _fileSystem;
+        public JsonReporter(StrykerOptions options, IFileSystem fileSystem)
         {
             _options = options;
+            _fileSystem = fileSystem;
         }
 
         public void OnAllMutantsTested(IReadOnlyInputComponent reportComponent)
         {
-            var jsonReport = JsonReportComponent.FromProjectComponent(reportComponent, _options.ThresholdOptions);
-            jsonReport.Name = _options.ProjectUnderTestNameFilter;
+            var jsonReport = JsonReportComponent.FromProjectComponent(reportComponent, _options);
+            jsonReport.ThresholdHigh = _options.ThresholdOptions.ThresholdHigh;
+            jsonReport.ThresholdLow = _options.ThresholdOptions.ThresholdLow;
+            jsonReport.ThresholdBreak = _options.ThresholdOptions.ThresholdBreak;
 
-            using (var file = File.CreateText(_options.BasePath + "/StrykerLogs/mutation-report.json"))
-            {
-                file.WriteLine(SerializeJsonReport(jsonReport));
-            }
+            WriteReportToJsonFile(jsonReport, Path.Combine(_options.BasePath, "/StrykerLogs/mutation-report.json"));
```

Comment:
Is StrykerLogs the correct location? For me `StrykerResults` sounds better. Also, if we add a timestamp to the filename more could be stored.

Reply:
perhaps, but I would prefer to then also change the log location to StrykerResults so we don't clutter too much